lightspark-sdk/src/commonTest/kotlin/com/lightspark/sdk/ClientIntegrationTests.kt

```diff
@@ -309,6 +309,44 @@ class ClientIntegrationTests {
         payments[0].id.shouldBe(payment?.id)
     }
 
+    @Test
+    fun `test getOutgoingPaymentsForPaymentHash`() = runTest {
+        val node = getFirstOskNode()
+        client.loadNodeSigningKey(node.id, PasswordRecoverySigningKeyLoader(node.id, NODE_PASSWORD))
+        val invoice = client.createTestModeInvoice(node.id, 100_000, "test invoice")
+        var outgoingPayment: OutgoingPayment? = client.payInvoice(node.id, invoice, maxFeesMsats = 100_000)
+        outgoingPayment.shouldNotBeNull()
+        while (outgoingPayment?.status == TransactionStatus.PENDING) {
+            delay(500)
+            outgoingPayment = OutgoingPayment.getOutgoingPaymentQuery(outgoingPayment.id).execute(client)
+            println("Payment status: ${outgoingPayment?.status}")
+        }
+        outgoingPayment?.status.shouldBe(TransactionStatus.SUCCESS)
+
+        val payments = client.getOutgoingPaymentForPaymentHash(outgoingPayment?.transactionHash!!)
```

Comment:
I still need to fix this one.  the invoice created on line 316 is just a string, is there a way to infer the payment has from that?

Reply:
One way to do this is to call decodeInvoice for that invoice, then you'll get the InvoiceData which contains paymentHash

https://github.com/lightsparkdev/kotlin-sdk/blob/6342b0fea7379b69fbb4141ea760cff92604d695/wallet-sdk/src/commonMain/kotlin/com/lightspark/sdk/wallet/model/InvoiceData.kt#L27

Reply:
Another way to do this without network request is to bech32 decode the invoice, and find the p field in it.

https://github.com/lightning/bolts/blob/master/11-payment-encoding.md#tagged-fields